docs/en/ingest-management/fleet/fleet-server-secrets.asciidoc

```diff
@@ -0,0 +1,265 @@
+----
+
+For Windows systems, there are no built-in options to create a RAM disk.
+If this is required, a third party program is needed.
+
+==== Example
+
+Here is a step by step guide for provisioning a service token on a Linux system:
+[source,sh]
+----
+sudo mkdir -p /mnt/fleet-server-secrets
+sudo mount -o size=1G -t tmpfs none /mnt/fleet-server-secrets
+echo -n MY-SERVICE-TOKEN > /mnt/fleet-server-secrets/service-token
+sudo chown root:root /mnt/fleet-server-secrets/service-token
+sudo chmod 0600 /mnt/fleet-server-secrets/service-token
+----
+
+NOTE: The `-n` flag is used with `echo` to prevent a newline character from being appended at the end of the secret. Care should be taken so that the secret file does not contain the trailing newline character.
+
+=== Secrets in Containers
+
+When using secret files directly in containers without using Kubernetes or another secrets management solution, the files can be passed into containers by mounting the file or directory.
+The file should be provisioned in the same manner as it is in <<secret-filesystem>> and mounted in read-only mode, for example when using Docker.
+
+If using {agent} image:
+[source,sh]
+----
+docker run \
+	-v /path/to/creds:/creds:ro \
+        -e FLEET_SERVER_CERT_KEY_PASSPHRASE=/creds/passphrase \
+        -e FLEET_SERVER_SERVICE_TOKEN_PATH=/creds/service-token \
+        --rm docker.elastic.co/beats/elastic-agent
+----
+
+=== Secrets in Kubernetes
+
+Kubernetes has a https://kubernetes.io/docs/concepts/configuration/secret/[secrets management engine] that can be used to provision secret files to pods.
+
+For example, you can create the passphrase secret with:
+[source,sh]
+----
+kubectl create secret generic fleet-server-key-passphrase \
+  --from-literal=value=PASSPHRASE
+----
+
+And create the service token secret with:
+[source,sh]
+----
+kubectl create secret generic fleet-server-service-token \
+  --from-literal=value=SERVICE-TOKEN
+----
+
+Then include it in the pod specification, for example, when you are running {fleet-server} under {agent}:
+[source,yaml]
+----
+spec:
+  volumes:
+  - name: key-passphrase
+    secret:
+      secretName: fleet-server-key-passphrase
+  - name: service-token
+    secret:
+      secretName: fleet-server-service-token
+  containers:
+  - name: fleet-server
+    image: docker.elastic.co/beats/elastic-agent
+    volumeMounts:
+    - name: key-passphrase
+      mountPath: /var/secrets/passphrase
+    - name: service-token
+      mountPath: /var/secrets/service-token
+    env:
+    - name: FLEET_SERVER_CERT_KEY_PASSPHRASE
+      value: /var/secrets/passphrase/value
+    - name: FLEET_SERVER_SERVICE_TOKEN_PATH
+      value: /var/secrets/service-token/value
+----
+
+==== {agent} Kubernetes secrets provider
+
+When you are running {fleet-server} under {agent} in Kuberenetes, {agent}'s <<kubernetes_secrets-provider>> may be used to insert a Kuberenetes secret directly into {fleet-server}'s configuration.
```

Comment:
```suggestion
When you are running {fleet-server} under {agent} in {k8s}, you can use {agent}'s <<kubernetes_secrets-provider>> to insert a {k8s} secret directly into {fleet-server}'s configuration.
```

Reply:
Tip: You can use the `(k8s}` attribute for quicker entry and to prevent spelling errors. 
FYI: It resolves here: https://github.com/elastic/docs/blob/master/shared/attributes.asciidoc#L252